File: 프로그래머스/KaKao/두 큐 합 같게 만들기.py
# 2022 KAKAO TECH INTERNSHIP 두 큐 합 같게 만들기
from collections import deque

def solution(queue1, queue2):
    answer = 0
    que1 = deque(queue1)
    que2 = deque(queue2)
    q1Tot = sum(queue1) # sum은 시간초과
    q2Tot = sum(queue2)
    
    # 큰넘걸 빼줌
    while True:
        # 답이 안나오는 경우
        # 두 큐가 다 돈 경우
        if answer > len(queue1)*4:
            return -1
        # 탈출 조건
        if q1Tot == q2Tot:
            break
        
        if q1Tot > q2Tot:
            ele = que1.popleft()
            que2.append(ele)
            q1Tot -= ele
            q2Tot += ele
            
        elif q1Tot < q2Tot:
            ele = que2.popleft()
            que1.append(ele)
            q1Tot += ele
            q2Tot -= ele

        answer += 1
    
    return answer

File: 프로그래머스/KaKao/두 큐 합 같게 만들기.swift
import Foundation

import Foundation

func solution(_ queue1:[Int], _ queue2:[Int]) -> Int {
    var answer: Int = 0
    
    var que1 = queue1
    var que2 = queue2
    var head1: Int = 0
    var head2: Int = 0
    
    var q1Sum = que1.reduce(0, +)
    var q2Sum = que2.reduce(0, +)
    let qSize = queue1.count * 4
    
    while head1 < qSize && head2 < qSize {
        if answer > qSize {
            answer = -1
            break
        }

        if q1Sum == q2Sum {
            break
        } else {
            if q1Sum > q2Sum {
                q1Sum -= que1[head1]
                q2Sum += que1[head1]
                que2.append(que1[head1])
                head1 += 1
            } else {
                q2Sum -= que2[head2]
                q1Sum += que2[head2]
                que1.append(que2[head2])
                head2 += 1
            }
            answer += 1
        }
    }
    
    return answer
}